src/DomCrawler/Field/InputFormField.php

```diff
@@ -22,6 +22,17 @@ final class InputFormField extends BaseInputFormField
 {
     use FormFieldTrait;
 
+    public function setValue($value)
+    {
+        if (\in_array($this->element->getAttribute('type'), ['text'], true)) {
```

Comment:
what about other textual types like `email` or `url` ?

Reply:
> ... 
> This PR did not take non-text types and html5 types into account.
I didn't want to enlarge this PR too much. 
I'll create an issue for more types and create a PR if I find some time.